ksqldb-common/src/main/java/io/confluent/ksql/errors/LogMetricAndContinueExceptionHandler.java

```diff
@@ -39,15 +41,40 @@ public DeserializationHandlerResponse handle(
   ) {
     log.debug(
         "Exception caught during Deserialization, "
-        + "taskId: {}, topic: {}, partition: {}, offset: {}",
+            + "taskId: {}, topic: {}, partition: {}, offset: {}",
         context.taskId(), record.topic(), record.partition(), record.offset(),
         exception
     );
 
     streamsErrorCollector.recordError(record.topic());
+
+    final OptionalLong restErrorCode = recursiveFindErrorCode(exception, 10);
+
+    if (restErrorCode.isPresent() && restErrorCode.getAsLong() == 40301) {
+      log.info(
+          "Permission error when attempting to access the schema during deserialization. "
+              + "taskId: {}, topic: {}, partition: {}, offset: {}",
+          context.taskId(), record.topic(), record.partition(), record.offset());
+      return DeserializationHandlerResponse.FAIL;
+    }
+
     return DeserializationHandlerResponse.CONTINUE;
   }
 
+  private OptionalLong recursiveFindErrorCode(final Throwable throwable, final int depthLeft) {
+
+    // depthLeft is used just here to safeguard against infinite recursion in the case of
+    // self-referencing exceptions
+    if (throwable != null && depthLeft > 0) {
+      if (throwable instanceof RestClientException) {
+        return OptionalLong.of(((RestClientException) throwable).getErrorCode());
+      }
+      return recursiveFindErrorCode(throwable.getCause(), depthLeft - 1);
+    }
```

Comment:
Would `Throwables.getRootCause(throwable)` from Guava help to get the `RestClientException` if found? 

Something like:
```
Throwable t = Throwables.getRootCause(throwable);
if (t instanceof RestClientException) {
   return ((RestClientException) t).getErrorCode() == FORBIDDEN_ERROR_CODE;
}
```

Reply:
Nice, I didn't know about `Throwables`. Mostly worked on projects with a lot fewer dependencies than KSQL ;). Since it's there already, lets use it, good idea.